crypto/ec/curve448/arch_arm_32/f_impl.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2014-2016 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#define GF_HEADROOM 2
+#define LIMB(x) (x##ull)&((1ull<<28)-1), (x##ull)>>28
+#define FIELD_LITERAL(a,b,c,d,e,f,g,h) \
+    {{LIMB(a),LIMB(b),LIMB(c),LIMB(d),LIMB(e),LIMB(f),LIMB(g),LIMB(h)}}
+
+#define LIMB_PLACE_VALUE(i) 28
+
+void gf_add_RAW(gf out, const gf a, const gf b)
+{
+    for (unsigned int i = 0; i < sizeof(*out) / sizeof(uint32xn_t); i++) {
```

Comment:
Minor nit, declare i outside of the for loop for pre C99 compatibility. Same in other places.

Reply:
Right - likely there are quite a few issues in the non 32-bit reference code (basically anything in one of the arch* directories that isn't arch_32). As I said in my intro to this, none of that code is actually used right now, and I simply imported it from Mike Hamburg's original contribution so we have the ability to hook it up later.

Reply:
Yep, sorry missed the 'arm' bit when reviewing this header, should have spotted it was code outside the review of just arch_32.